src/buf/fixed/pool.rs

```diff
@@ -274,8 +274,10 @@ impl<T: IoBufMut> FixedBufPool<T> {
         pin!(notified);
         loop {
             // In the single-threaded case, no buffers could get checked in
-            // between us calling `try_next` and here, so we can't miss a wakeup.
-            notified.as_mut().await;
+            // between us calling `try_next` and here. However, we may still miss a wake-up,
+            // as multiple check-ins can occur before any waking tasks are scheduled,
+            // which would result in the loss of a permit
```

Comment:
This is right, thanks for the clarification. In fact, the preamble of this comment can be completely replaced because it's irrelevant: multiple buffers can still be checked in while awaiting on `notified` before another iteration of the loop.